adr/ADR-8.md

```diff
@@ -293,6 +303,60 @@ all the known services we ask for all subjects below `$KV.SERVICES.nats.>`. A no
 
 This would mean many calls to the underlying API call for known subjects as we would not know via a typical watch about expired data.
 
+#### API Design notes
+
+The API here represents a minimum, languages can add local flavour to the API - for example one can add `PutUint64()` and `GetUint64()`
+in addition to the `Get()` and `Put()` defined here, but it's important that as far as possible all implementations strive to match
+the core API calls - `Get()`, `Put()` and to a lessor extend `Delete()` and `Purge()` and `Entry` - the rest, admin APIs, and supporting
+calls can be adjusted to be a natural fit in the language and design.  This is in line with existing efforts to harmonize 
+`Subscribe()`, `Publish()` and more.
+
+The design is based on extensive prior art in the industry. During development 20+ libraries in Go, Ruby, Python and Java were reviewed 
+for their design choices and this influenced things like `Get()` returning an `Entry`.
+
+Consul Go:
+
+```go
+func (k *KV) Put(p *KVPair, q *WriteOptions) (*WriteMeta, error){}
+func (k *KV) Get(key string, q *QueryOptions) (*KVPair, *QueryMeta, error)
+```
+
+Here KVPair has properties like `Key`, various sequences, flags, timestamps etc and `Value []byte`.
+
+Dynanic language libraries like Python for example return maps, but there too the return value of a default `get()` operation is a hash
+with all metadata and value stored in `Value` or similar.
+
```

Comment:
hmm. Maybe I'm looking at the wrong thing: https://docs.python.org/3/tutorial/datastructures.html#dictionaries